butane_core/src/migrations/adb.rs

```diff
@@ -318,6 +328,45 @@ impl From<TypeIdentifier> for DeferredSqlType {
     }
 }
 
+/// Abstract representation of a database column reference constraint.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub enum ARef {
+    Literal(ARefLiteral),
+    Deferred(DeferredSqlType),
```

Comment:
very likely, this can be simply `Deferred,` without any data, as I suspect that the referred to column is always a available in the column `sqltype`

Reply:
IMO this isnt worth entwining this into the `resolve_types` process.